### PR TITLE
feat: reconcile paper fills to trade idea audits

### PR DESCRIPTION
**Pull request by Solders-Girdles**
## Summary
Related issue / finding / RJ-routed package: Closes #1035

Adds a paper/mock fill reconciliation bridge for approved trade ideas without changing live execution. The new `PaperFillReconciler` reads existing persisted `EventStore` trade events, matches filled paper/mock orders by explicit `decision_id`, `client_order_id`, or a conservative unique approved symbol/side fallback, and records matched submissions/fills only through `TradeIdeaService`. Unmatched, ambiguous, terminal, or duplicate fills are reported without mutation.

## Type of change
- [ ] Refactor
- [x] Feature
- [ ] Bug fix
- [x] Tests only
- [x] CI/Docs/Tooling

## Scope & Impact
- Affected areas / components: `features/trade_ideas`, `ideas` CLI, generated `var/agents/testing/**` inventory.
- Behavior changes: new opt-in `gpt-trader ideas reconcile-paper-fills` command. It defaults to dry-run, requires `--apply` to append audit events, and rejects non-paper/dev/mock profiles before reading runtime persistence.
- No live Coinbase/account reads, no order submission, no approval automation, no policy changes, and no closeout mutation.

## Test Plan
- [x] Unit tests added/updated
- [ ] Integration tests added/updated
- [ ] Contract tests (if applicable)
- [ ] Ran locally: `pytest -m "unit and not slow"`
- [x] Markers used appropriately (`unit`/`integration`/`slow`/`perf`)

Verification run:
- `uv run pytest tests/unit/gpt_trader/features/trade_ideas/test_paper_reconciliation.py tests/unit/gpt_trader/cli/commands/test_ideas_paper_reconciliation.py tests/unit/gpt_trader/cli/commands/test_ideas_manual_records.py tests/unit/gpt_trader/features/trade_ideas/test_service_venue_boundary.py -q` -> 12 passed
- `uv run ruff check src/gpt_trader/features/trade_ideas/paper_reconciliation.py src/gpt_trader/features/trade_ideas/__init__.py src/gpt_trader/cli/commands/ideas.py tests/unit/gpt_trader/features/trade_ideas/test_paper_reconciliation.py tests/unit/gpt_trader/cli/commands/test_ideas_paper_reconciliation.py` -> passed
- `uv run mypy src/gpt_trader/features/trade_ideas/paper_reconciliation.py src/gpt_trader/cli/commands/ideas.py` -> passed
- `uv run agent-regenerate --verify` -> up to date
- `git diff --check` -> passed

Skipped by design:
- `broker-check`, production preflight, canary readiness, live trading, order-affecting commands, and strict/full local CI.

## Complexity & Quality Gates
- [x] Mypy/type-check passed
- [x] Xenon/radon complexity gate passed (CC <= 15 on live_trade execution) - not applicable; no live_trade execution code changed
- [x] No `sys.path` hacks in tests
- [x] No `MagicMock` on `async def` (use `AsyncMock`)
- [x] Import-lint rules respected (no “import up the stack”)

## Observability & Errors
- [ ] Structured JSON logs for new/changed paths
- [x] Errors include diagnostic context (symbol, order_id, correlation_id)
- [ ] Added/updated sampling examples in tests (if applicable)

The CLI returns structured JSON/text reconciliation reports. Live-profile rejection uses the existing `ValidationError` path with `field=profile`; unmatched fill entries include event order/client ids and reason strings.

## Configuration
- [ ] If config changed: `python -m gpt_trader.cli.config validate --profile <name>` passes
- [ ] Env docs re-generated (if applicable) and committed
- [x] No `ConfigLoader` usages introduced (ConfigManager-only)

No runtime profile/config files changed. The CLI profile gate is command-local and accepts only `paper`, `dev`, or `mock`.

## Breaking Changes
- [x] None
- [ ] Documented in PR body and release notes if any

## Screenshots / Logs (optional)
N/A

## Checklist
- [x] Acceptance criteria in linked issue(s), finding packet, or routed package met
- [x] Affected paths listed in PR description
- [ ] Changelog entry (if repo uses one)

## Summary by CodeRabbit

* **New Features**
  * Added an `ideas reconcile-paper-fills` command to reconcile persisted paper/dev/mock filled events against approved trade ideas, with profile validation and configurable actor/event-store root.
  * Provides JSON reporting with matched/unmatched/skipped details for dry-run and `--apply`, including audit/venue stamping.
* **Bug Fixes**
  * Improved match selection to avoid unintended fallback and handle symbol/side/client order conflicts safely.
  * Prevents duplicate re-application and skips already-terminal/previously recorded fills.
* **Tests**
  * Added unit coverage for CLI dry-run vs apply, validation failures, event-store root precedence, and reconciliation edge cases.

### CODE REVIEW

**Comment by Solders-Girdles** [comment]
@codex review

Addressed:
- `discussion_r3486121785`: dry-run reconciliation now maintains an in-memory preview of filled decisions and refreshes symbol/side candidates between events, so a direct client-order/decision-id fill followed by a legacy same-symbol/side fill previews the same two matched decisions that `--apply` records.

Verification:
- `uv run pytest tests/unit/gpt_trader/features/trade_ideas/test_paper_reconciliation.py tests/unit/gpt_trader/cli/commands/test_ideas_paper_reconciliation.py -q` -> passed, 20 tests.
- `uv run ruff check src/gpt_trader/features/trade_ideas/paper_reconciliation.py src/gpt_trader/cli/commands/ideas.py tests/unit/gpt_trader/features/trade_ideas/test_paper_reconciliation.py tests/unit/gpt_trader/cli/commands/test_ideas_paper_reconciliation.py` -> passed.
- `uv run mypy src/gpt_trader/features/trade_ideas/paper_reconciliation.py src/gpt_trader/cli/commands/ideas.py` -> passed.
- `uv run agent-regenerate --verify` -> passed after regenerating testing metadata.
- `uv run black --check src/gpt_trader/features/trade_ideas/paper_reconciliation.py tests/unit/gpt_trader/features/trade_ideas/test_paper_reconciliation.py` -> passed.
- `git diff --check` -> passed.

File: src/gpt_trader/features/trade_ideas/paper_reconciliation.py [hunk 1]
```diff
@@ -0,0 +1,446 @@
...
+
+
+def _first_existing_decision_id(
+    event: PaperFillEvent,
+    *,
+    view_by_decision_id: Mapping[str, TradeIdeaView],
+) -> str | None:
+    if event.decision_id and event.decision_id in view_by_decision_id:
+        return event.decision_id
+    return None
```
[the remark this answers is not in the document]
Comment (Solders-Girdles) [reply src/gpt_trader/features/trade_ideas/paper_reconciliation.py]: Addressed on this branch: an explicit `decision_id` absent from the current views now reports unmatched immediately instead of falling back to client-order/symbol-side matching.

File: src/gpt_trader/cli/commands/ideas.py [hunk 2]
```diff
@@ -522,6 +572,19 @@ def _actor_id(args: Namespace) -> str:
...
+    explicit_actor = getattr(args, "actor", None)
+    return resolve_trade_idea_actor_id(explicit_actor or "paper-fill-reconciler")
+
+
+def _paper_reconciliation_event_store_root(args: Namespace, profile: str) -> Path:
+    configured_root = getattr(args, "event_store_root", None)
+    if configured_root is not None:
+        return cast(Path, configured_root)
+    profile_root = "dev" if profile == "mock" else profile
+    return Path("runtime_data") / profile_root
```
[the remark this answers is not in the document]
Comment (Solders-Girdles) [reply src/gpt_trader/cli/commands/ideas.py]: Addressed on this branch: `ideas reconcile-paper-fills` resolves an omitted `--event-store-root` through `BotConfig.from_env()`/`resolve_runtime_paths`; explicit roots still win.

File: src/gpt_trader/features/trade_ideas/paper_reconciliation.py [hunk 3]
```diff
@@ -0,0 +1,488 @@
...
+        apply: bool = False,
+    ) -> PaperFillReconciliationReport:
+        """Reconcile normalized paper fill events."""
+        views = tuple(self._service.list_views())
+        view_by_decision_id = {view.idea.decision_id: view for view in views}
+        recordable_views = tuple(
+            view
+            for view in views
+            if view.state in {TradeIdeaState.APPROVED, TradeIdeaState.SUBMITTED}
+        )
```
[the remark this answers is not in the document]
Comment (Solders-Girdles) [reply src/gpt_trader/features/trade_ideas/paper_reconciliation.py]: Addressed on this branch: applied fills refresh the recordable symbol/side candidates between events so a filled idea no longer blocks a later legacy same-symbol/side fill.

File: src/gpt_trader/features/trade_ideas/paper_reconciliation.py [hunk 4]
```diff
@@ -0,0 +1,491 @@
...
+                        match_method=method,
+                        final_state=view.state.value,
+                    )
+                )
+                continue
+
+            recorded_submission = False
+            recorded_fill = False
+            final_state = view.state.value
+            if apply:
```
[the remark this answers is not in the document]
Comment (Solders-Girdles) [reply src/gpt_trader/features/trade_ideas/paper_reconciliation.py:312]: Addressed on this branch: dry-run now records filled decisions in `preview_filled_decision_ids` and refreshes candidates between events, so the preview matches `--apply`.

File: src/gpt_trader/features/trade_ideas/paper_reconciliation.py [hunk 5]
```diff
@@ -0,0 +1,518 @@
...
+        return None, None, "no approved idea matched fill"
+
+    def _already_reconciled(self, view: TradeIdeaView, event: PaperFillEvent) -> bool:
+        external_order_id = event.external_order_id
+        if not external_order_id:
+            return view.state is TradeIdeaState.FILLED
+        return any(
+            audit_event.action is AuditAction.FILLED
+            and audit_event.external_order_id == external_order_id
+            for audit_event in view.events
```
[the remark this answers is not in the document]
Comment (Solders-Girdles) [reply src/gpt_trader/features/trade_ideas/paper_reconciliation.py:434]: Fixed in `f43934f3`: `_already_reconciled` now checks external order ids recorded on every filled audit event (and on fills applied within the run), so a rerun over the full store can't re-record an id-less legacy fill onto a newer same-symbol/side idea. Regression test added.

File: src/gpt_trader/features/trade_ideas/paper_reconciliation.py [hunk 6]
```diff
@@ -0,0 +1,536 @@
...
+            event.client_order_id,
+            view_by_decision_id=view_by_decision_id,
+        )
+        if client_match is not None:
+            return client_match, "client_order_id", "matched client_order_id"
+
+        symbol_side_matches = [
+            view
+            for view in recordable_views
+            if _symbol_matches(view, event) and _side_matches(view, event)
```
[the remark this answers is not in the document]
Comment (Solders-Girdles) [reply src/gpt_trader/features/trade_ideas/paper_reconciliation.py]: Fixed in `03f97615`: the symbol/side fallback now gates already-submitted ideas so they only match a legacy fill whose external id equals their submitted order id (submitted-as-A / filled-as-B is no longer possible). Regression test added.

File: src/gpt_trader/features/trade_ideas/paper_reconciliation.py [hunk 7]
```diff
@@ -0,0 +1,562 @@
...
+        )
+        if client_match is not None:
+            return client_match, "client_order_id", "matched client_order_id"
+
+        symbol_side_matches = [
+            view
+            for view in recordable_views
+            if _symbol_matches(view, event)
+            and _side_matches(view, event)
+            and _submission_order_id_allows(view, event)
```
[the remark this answers is not in the document]
Comment (Solders-Girdles) [reply src/gpt_trader/features/trade_ideas/paper_reconciliation.py:401]: Fixed in `5c3835d5`: a non-empty `client_order_id` that does not resolve now reports unmatched instead of falling back to symbol/side, so a stale/moved client id can't attach to an unrelated same-symbol/side idea. The symbol/side fallback is now reachable only for fills with neither a decision id nor a client id. Regression test added.

File: src/gpt_trader/features/trade_ideas/paper_reconciliation.py [hunk 8]
```diff
@@ -0,0 +1,562 @@
...
+        audited_fill_order_ids: AbstractSet[str],
+    ) -> bool:
+        external_order_id = event.external_order_id
+        if not external_order_id:
+            return view.state is TradeIdeaState.FILLED
+        # An order id recorded against *any* filled idea means this fill has
+        # already been reconciled. Checking only the matched view would let a
+        # rerun over the full event store re-record a legacy (id-less) fill
+        # against a newer same-symbol/side idea and corrupt its audit trail.
+        return external_order_id in audited_fill_order_ids
```
[the remark this answers is not in the document]
Comment (Solders-Girdles) [reply src/gpt_trader/features/trade_ideas/paper_reconciliation.py]: Fixed in `99bda57d`: the global order-id dedupe is now restricted to legacy (id-less, symbol/side-matched) fills. Explicitly id-matched fills (decision/client id) dedupe only against their own matched idea's audit trail, so a reused `MOCK_000001` from a DeterministicBroker counter reset carrying a new client id is no longer skipped. Regression test added.
